store/postgres/src/relational_queries.rs

```diff
@@ -473,8 +474,32 @@ impl EntityData {
                     // column; those will be things like the block_range that
                     // is used internally for versioning
                     if key == "g$parent_id" {
-                        let value = T::Value::from_column_value(&ColumnType::String, json)?;
-                        out.insert_entity_data("g$parent_id".to_owned(), value);
+                        match &parent_type {
+                            None => {
+                                if ENV_VARS.store.disable_error_for_toplevel_parents {
+                                    // Only temporarily in case reporting an
+                                    // error causes unexpected trouble. Can
+                                    // be removed once it's been working for
+                                    // a few days
+                                    let value =
+                                        T::Value::from_column_value(&ColumnType::String, json)?;
+                                    out.insert_entity_data("g$parent_id".to_owned(), value);
+                                } else {
+                                    // A query that does not have parents
+                                    // somehow returned parent ids. We have no
+                                    // idea how to deserialize that
+                                    if parent_type.is_none() {
+                                        return Err(graph::constraint_violation!(
+                                            "query unexpectedly produces parent ids"
+                                        ));
+                                    }
```

Comment:
Why do we need this check? I think it's redundant, at this point in the code we already know that `parent_type.is_none()`.

Reply:
Yes, that's absolutely right. Fixed.